### PR TITLE
Fix @:using for interfaces and abstracts

### LINKED ISSUE #8188: @:using should work for all type kinds
Just merging these three issues that are all essentially about the same thing: `@:using` not behaving as expected and only working on some kinds of types.

- #7859
- #7930
- #8022

### PR DESCRIPTION
**Pull request by RealyUniqueName**
#8188

### CODE REVIEW

**Comment by RealyUniqueName** [comment]
@Simn ready for review

File: src/typing/typeloadFields.ml [hunk 1]
```diff
@@ -576,7 +591,13 @@ let build_fields (ctx,cctx) c fields =
 	let get_fields() = !fields in
 	let pending = ref [] in
 	c.cl_build <- (fun() -> BuildMacro pending);
-	build_module_def ctx (TClassDecl c) c.cl_meta get_fields cctx.context_init (fun (e,p) ->
+	let meta =
```
Comment (Simn) [inline src/typing/typeloadFields.ml]: Can you try if instead of doing this we can add `Meta.Using` to the list of metas at typeloadModule.ml line 325?

**Comment by Gama11** [comment]
Since the title says "for interfaces and abstracts", I take it the verdict is to not support it for typedefs / structures?

In that case, can we have a compiler error when trying to use the meta there?

**Comment by RealyUniqueName** [comment]
Actually, I forgot, we need to decide on typedefs :)
I'm ok with any option - either allow or prohibit `@:using` on typedefs.
@Simn @ncannasse ?

**Comment by Simn** [comment]
Can we worry about that later and merge this first?

**Pull request merged**